tests/network/primary_pod_network.go

```diff
@@ -85,60 +75,106 @@ var _ = SIGDescribe("Primary Pod Network", func() {
 			var vmi *v1.VirtualMachineInstance
 
 			BeforeEach(func() {
-				vmi = setupVMI(virtClient, vmiWithMasqueradeBinding())
+				vmi = libvmi.SetupVMI(vmiWithMasqueradeBinding(), tests.LoggedInCirrosExpecter)
 			})
 
 			AfterEach(func() {
-				cleanupVMI(virtClient, vmi)
+				libvmi.CleanupVMI(vmi)
 			})
 
 			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
 		})
 	})
-})
 
-func setupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
-	By("Creating the VMI")
-	var err error
-	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
-	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+	Describe("Connectivity", func() {
+		const servicePort = 1500
+		const ipv4HeaderSize = 20
+		const icmpHeaderSize = 8
+		const ipv4IcmpHeaderSize = ipv4HeaderSize + icmpHeaderSize
 
-	By("Waiting until the VMI gets ready")
-	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+		AssertConnectivity := func(outboundVMI, inboundVMI *v1.VirtualMachineInstance) {
+			inboundVMIAddress := inboundVMI.Status.Interfaces[0].IP
 
-	return vmi
-}
+			By("Checking connectivity via ICMP")
+			Expect(tests.PingFromVMConsole(outboundVMI, inboundVMIAddress)).To(Succeed())
```

Comment:
ExpectWithOffset please, there are more like that

Reply:
In this case, I don't agree. The `It` has only a single line executing this function. If the test fails, knowing that its single line failed brings no value to me. i want to know where did it fail in this Assert function which contains the whole body of the test.

Reply:
`AssertConnectivity` is called by more than one caller, and in the future more may use it. In such a case, it is harder to understand the traceback. Going directly to the line number is preferable, so I think either return the error or assert with an offset is preferable.

If we cannot return the error and assert in the caller, then I do not feel this is a blocker.